governance/program/src/tools/token.rs

```diff
@@ -3,17 +3,45 @@
 use arrayref::array_ref;
 use solana_program::{
     account_info::AccountInfo,
+    decode_error::DecodeError,
     entrypoint::ProgramResult,
     msg,
     program::{invoke, invoke_signed},
-    program_error::ProgramError,
+    program_error::{PrintProgramError, ProgramError},
     program_pack::Pack,
     pubkey::Pubkey,
     rent::Rent,
     system_instruction,
 };
 
-use crate::error::GovernanceError;
+use num_derive::FromPrimitive;
+use thiserror::Error;
+
+/// SPL Token Tools errors  
+#[derive(Clone, Debug, Eq, Error, FromPrimitive, PartialEq)]
+pub enum SplTokenToolsError {
+    /// Invalid Token account owner
+    #[error("Invalid Token account owner")]
+    InvalidTokenAccountOwner,
+}
+
+impl PrintProgramError for SplTokenToolsError {
+    fn print<E>(&self) {
+        msg!("SPL-TOKEN-TOOLS-ERROR: {}", &self.to_string());
+    }
+}
+
+impl From<SplTokenToolsError> for ProgramError {
+    fn from(e: SplTokenToolsError) -> Self {
+        ProgramError::Custom(e as u32)
```

Comment:
When you have multiple error types, because of this line, they'll get all mixed up in the logs, so you'll be better off keeping them all in one enum, and printing a specific message before returning them.

This is already an issue when programs propagate error codes from the token program, and get confused because they're not sure if it's an error from their program or the token program.

Reply:
This is unfortunate, I wanted to keep it generic. 
I moved the errors back to GovernanceError